File: app/src/main/java/com/example/javatokotlin/javaCode/MainActivity.java
package com.example.javatokotlin.javaCode;

import android.content.res.Resources;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;
import com.example.javatokotlin.R;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

    Button resetButton;
    Button countButton;
    Button randomButton;
    TextView numberTextView;
    ConstraintLayout layout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        layout = findViewById(R.id.constraintLayout);
        resetButton = findViewById(R.id.resetButton);
        countButton = findViewById(R.id.countButton);
        randomButton = findViewById(R.id.randomButton);
        numberTextView = findViewById(R.id.numberTextView);

        initUi();
    }

    private void initUi() {
        numberTextView.setText("0");

        countButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                countMe(numberTextView);
            }
        });

        resetButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                numberTextView.setText("0");
                updateBackgroundColor(0);
            }
        });

        randomButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Integer randomInt = randomInt(50);
                numberTextView.setText(randomInt.toString());
                updateBackgroundColor(randomInt);
            }
        });
    }

    private Integer randomInt(Integer maximum) {
        Random random = new Random();
        return random.nextInt(maximum) + 1;
    }

    private void countMe(TextView textView) {
        String countString = textView.getText().toString();
        Integer count = Integer.parseInt(countString);
        count++;
        updateBackgroundColor(count);
        textView.setText(count.toString());
    }

    private void updateBackgroundColor(Integer number) {
        Resources resource = getResources();
        Integer colorRes;
        if (number > 0 && number <= 10) {
            colorRes = resource.getColor(android.R.color.holo_red_dark);
        } else if (number > 10 && number <= 20) {
            colorRes = resource.getColor(android.R.color.holo_green_dark);
        } else if (number > 20 && number <= 30) {
            colorRes = resource.getColor(android.R.color.holo_blue_dark);
        } else if (number > 30 && number <= 40) {
            colorRes = resource.getColor(android.R.color.holo_orange_dark);
        } else if (number > 40 && number <= 50) {
            colorRes = resource.getColor(android.R.color.holo_purple);
        } else {
            colorRes = resource.getColor(android.R.color.black);
        }
        layout.setBackgroundColor(colorRes);
    }

}

File: app/src/main/java/com/example/javatokotlin/MainActivity.kt
package com.example.javatokotlin

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.preference.PreferenceManager
import android.view.Menu
import android.view.MenuItem
import android.widget.TextView

import androidx.appcompat.app.AppCompatActivity
import com.example.javatokotlin.extention.getColorCompat
import com.example.javatokotlin.extention.toast
import kotlinx.android.synthetic.main.activity_main.*

import java.util.Random

class MainActivity : AppCompatActivity() {

    companion object {
        const val FIRST_RANGE_COUNTER = "first_range_counter"
        const val SECOND_RANGE_COUNTER = "second_range_counter"
        const val THIRD_RANGE_COUNTER = "third_range_counter"
        const val FOURTH_RANGE_COUNTER = "fourth_range_counter"
        const val FIFTH_RANGE_COUNTER = "fifth_range_counter"
        const val OTHER_COUNTER = "other_counter"
    }

    private val defaultNumber: String by lazy { randomInt(50).toString() }
    private lateinit var sharePreference: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        sharePreference = PreferenceManager.getDefaultSharedPreferences(this)
        initUi()
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_stat, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.stat_screen) {
            navigateToSecondActivity()
        }
        return super.onOptionsItemSelected(item)
    }

    private fun navigateToSecondActivity() {
        startActivity(Intent(this, SecondActivity::class.java))
    }

    private fun initUi() {
        numberTextView.text = defaultNumber
        updateBackgroundColor(defaultNumber.toInt())
        resetButton.setOnClickListener {
            numberTextView.text = defaultNumber
            updateBackgroundColor(defaultNumber.toInt())
        }
        countButton.setOnClickListener { countMe(numberTextView) }
        randomButton.setOnClickListener {
            val randomInt = randomInt(50)
            numberTextView.text = randomInt.toString()
            updateBackgroundColor(randomInt)
        }
    }

    private fun randomInt(maximum: Int) = Random().nextInt(maximum) + 1

    private fun countMe(textView: TextView) {
        val countString = textView.text.toString()
        var count: Int = Integer.parseInt(countString)
        count++
        updateBackgroundColor(count)
        textView.text = count.toString()
        toast("before : $countString")
    }

    private fun updateBackgroundColor(number: Int) {
        constraintLayout.setBackgroundColor(when (number) {
            in 1..10 -> {
                increaseCounter(FIRST_RANGE_COUNTER)
                getColorCompat(android.R.color.holo_red_dark)
            }
            in 11..20 -> {
                increaseCounter(SECOND_RANGE_COUNTER)
                getColorCompat(android.R.color.holo_green_dark)
            }
            in 21..30 -> {
                increaseCounter(THIRD_RANGE_COUNTER)
                getColorCompat(android.R.color.holo_blue_dark)
            }
            in 31..40 -> {
                increaseCounter(FOURTH_RANGE_COUNTER)
                getColorCompat(android.R.color.holo_orange_dark)
            }
            in 41..50 -> {
                increaseCounter(FIFTH_RANGE_COUNTER)
                getColorCompat(android.R.color.holo_purple)
            }
            else -> {
                increaseCounter(OTHER_COUNTER)
                getColorCompat(android.R.color.black)
            }
        })
    }

    private fun increaseCounter(key: String) {
        with(sharePreference) {
            var counter = getInt(key, 0)
            counter++
            edit().putInt(key, counter).apply()
        }

    }
}